[GALLERY] Gallery landing page

bentranter opened the pull request:
This PR adds the Gallery landing page, along with its associated styles and components.

In this particular PR there are dead links to a template named `gallery-inner.amp.html`, but those will be added in the next PR.

I know there are a fair number of changed files here, but the majority are images that we'll need for this PR as well as the next one, so hopefully the actual amount of code here isn't too big for a review :)

bentranter commented:
I signed it! (this account should be associated with our org's CLA account)

camelburrito commented on tasks/validate.js:
```diff
@@ -20,8 +20,8 @@ const config = require('./config');
   return gulp.src([
-        `${config.dest.templates}/**/*.html`,
-        `!${config.dest.hl_partials}/**/*.html`,
+        `${config.dest.templates}/**/*amp.html`,
```
pls dont change any existing conventions. You will need to do this in a separate PR and have it reviewed and we will need to make sure it does not break anything else.

camelburrito commented on templates/gallery/partials/header.snip.html:
```diff
@@ -0,0 +1,47 @@
+-->
+}}
+
+{{#header}}
```
please reuse navbar.

bentranter replied:
Looking into this right now, should have an update about it tomorrow.

camelburrito commented on templates/gallery/partials/sidebar.snip.html:
```diff
@@ -0,0 +1,75 @@
+-->
+}}
+
+{{#sidebar}}
```
reuse navbar

bentranter replied:
We just use this to wrap `<amp-sidebar>`. Is that okay?

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+      </h1>
+      <amp-carousel width="100vw" height="100vh" controls class="ampstart-gallery-landing-wall block relative">
+        <div class="ampstart-gallery-landing-wall-scroll relative">
+          <div class="ampstart-gallery-landing-image-wrap-0" >
```
why do we need the nested divs? - also this is not a good candidate for a class , why are we using number suffixes? you can do this is css using n-th child and odd even rules?

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+      <amp-carousel width="100vw" height="100vh" controls class="ampstart-gallery-landing-wall block relative">
+        <div class="ampstart-gallery-landing-wall-scroll relative">
+          <div class="ampstart-gallery-landing-image-wrap-0" >
+            <div class="ampstart-gallery-landing-image-transform">
```
try not to nest un-necessary wrapping divs , i guess a better alternative could be ::before and ::after?

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+      <h1 class="ampstart-gallery-landing-title flex justify-center items-center center absolute top-0 left-0 right-0 z1 mx-auto ampstart-gallery-base-invert">
+        {{{header.heading}}}
+      </h1>
+      <amp-carousel width="100vw" height="100vh" controls class="ampstart-gallery-landing-wall block relative">
```
do not prefix gallery classes with ampstart- 

the ampstart- prefix should only be used on ampstart- classes. 

You will need to come up with a name for your namespace and start using it. I think it is going to be mueum?

bentranter replied:
`ampstart` prefixes removed and replaced with `gallery`, since that's what we'd been using, and the directory is named `gallery` as well

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+                srcset="
+                  ../../img/gallery/548241_200.jpg 1x,
+                  ../../img/gallery/548241_400.jpg 2x
+                "
```
alt="some nice decsription" on all images. (both here and everywhere else)

bentranter replied:
Fixed in latest changes

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+                height="12"
+                layout="responsive"
+                noloading
+                src="../../img/gallery/548241_200.jpg"
```
please give a human readable name for all images.(both here and everywhere else)

bentranter replied:
Fixed in latest changes

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+                "
+              ></amp-img>
+            </div>
+            <div class="ampstart-gallery-landing-image-caption ampstart-gallery-landing-image-caption-bottom ampstart-gallery-landing-image-caption-extraleft">
```
overall i think there is a lack of semantics here and everywhere else 

https://www.iandevlin.com/blog/2011/04/html5/html5-section-or-article

Please use semantics in HTML , plain divs are not good everywhere. 
http://html5doctor.com/lets-talk-about-semantics/

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+            </div>
+            <div class="ampstart-gallery-landing-image-caption ampstart-gallery-landing-image-caption-bottom ampstart-gallery-landing-image-caption-extraleft">
+              <div class="ampstart-gallery-landing-image-caption-content">
+                Modern Caesar<br> Sylvia Johnson
```
isnt there already an ampstart component for image with caption? please reuse.

bentranter replied:
The ampstart component for an image with a caption contains a hardcoded image, and generally suit the styles we're trying to apply here. I'll push a staging branch of this PR for reference

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+            <div class="ampstart-gallery-landing-image-caption ampstart-gallery-landing-image-caption-bottom ampstart-gallery-landing-image-caption-extraleft">
+              <div class="ampstart-gallery-landing-image-caption-content">
+                Modern Caesar<br> Sylvia Johnson
+                <br><a href="./gallery-inner.amp.html" class="ampstart-gallery-landing-caption-link inline-block">View Exhibit</a>
```
no brs please.

bentranter replied:
Fixed

camelburrito commented on templates/gallery/gallery.amp.html:
```diff
@@ -0,0 +1,434 @@
+              </div>
+            </div>
+          </div>
+          <div class="ampstart-gallery-landing-image-wrap-1">
```
i see this pattern repeating - move it to a partial?

bentranter replied:
Updated in latest PR, this is moved to a `landing-images` partial

10xjs commented on css/templates/gallery/v-hide.css:
```diff
@@ -0,0 +1,33 @@
+@import './bass-override-vars';
+
+@media (--breakpoint-v-xs) {
+  .v-xs-hide { display: none }
```
This and other "v-" breakpoints measure device height instead of width. I used them for styling the vertical elements at the left of the design.

10xjs commented on gulpfile.js:
```diff
@@ -200,8 +203,8 @@ gulp.task('postcss', 'build postcss files', function() {
     require('postcss-calc')(),
-    require('postcss-color-function')(),
     require('postcss-custom-properties')(),
+    require('postcss-color-function')(),
```
We aren't adding this, only changing its order in the list. It must be processed after `custom-properties` to enable statements like `color(var(--blue) a(0.5))`.

camelburrito commented:
@bentranter - let me know when your next PR (or resolution this PR is ready for review.) Let me know if you need some help!

bentranter commented on css/templates/gallery/bass-override-vars.css:
```diff
@@ -0,0 +1,71 @@
+/**
```
Fixed

bentranter commented on gulpfile.js:
```diff
@@ -134,7 +134,7 @@ gulp.task('templateapi', function() {
 });
 
-function inlineCheckScript(node) {
+function disableNode(node) {
```
This was a mistake, it's removed now

bentranter commented:
Majority of comments are now addressed, also I've created staging branch for this (and future) PRs: https://ampstart-pr-metalab.firebaseapp.com/templates/gallery/gallery.amp

Regarding the reuse of the `navbar` component for the sidebar and header, I feel like it's not worth it due to how much we'd need to override. Please see the staging link if you have a chance, it should highlight how drastically different our header and sidebar is from the `navbar` component.

Regarding semantics, I've updated a few of the tags from `<div>` to something more descriptive. I've also refactored the majority of the landing page into a component, so that should help eliminate a lot of the repetitive markup.

Finally, all unused CSS was removed from this PR, as well as the changes to the validation task and gulpfile were reverted.

camelburrito commented:
@bentranter 

> Regarding the reuse of the navbar component for the sidebar and header, I feel like it's not worth it due to how much we'd need to override. 

I just took a look and i think it is totally reusable. Not sure what is blocking. 

I think it might be easier if you pull the navbar in a separate PR and we can work on it fast, Ill take a pass at rest of the stuff soon.

camelburrito commented on css/templates/gallery/components/_header.css:
```diff
@@ -0,0 +1,149 @@
+ * limitations under the License.
+ */
+
+.gallery-component-header {
```
i guess just call it `gallery-header`?

camelburrito commented:
@bentranter - Are we still proceeding with this PR?

bentranter commented on css/templates/gallery/components/_landing-images.css:
```diff
@@ -0,0 +1,570 @@
+  pointer-events: none;
+}
+
+.gallery-landing-image-wrap-0 amp-img { background: #b3afab }
```
I believe this was for selector performance -- selecting a single class is (very roughly) 50 to 100 milliseconds faster than an nth-child selector (as discovered using [this test](https://benfrain.com/selector-test/index.html). Since we have 18 of these, we end up saving 1 to 2 seconds on rendering just from this change. I can change it to use nth child if you'd prefer.

bentranter commented on css/templates/gallery/page-vars.css:
```diff
@@ -0,0 +1,81 @@
+
+  --bold-font-weight: bold;
+
+  --width-1: 24rem;
```
Yup, these are rems

bentranter commented on css/templates/gallery/components/_landing-images.css:
```diff
@@ -0,0 +1,570 @@
+.gallery-landing-image-wrap-16 amp-img { background: #F2BFCD }
+.gallery-landing-image-wrap-17 amp-img { background: #E3CB78 }
+
+[class^="gallery-landing-image-wrap-"]:hover amp-img {
```
See my comment above about this one too

bentranter commented on css/templates/gallery/page-vars.css:
```diff
@@ -0,0 +1,81 @@
+ * limitations under the License.
+ */
+
+:root {
```
Not sure what you mean here -- where do you think they are duplicated?

bentranter commented:
Hey @camelburrito, yup, we're still proceeding with this PR. I've made a ton of changes based on feedback I received from my team (that I believe came from you a while ago). If you could take another pass at this, that would be much appreciated.

bentranter commented:
Closing in favour #646
